### PR TITLE
solution

### PR DESCRIPTION
**Pull request by nrjss67**

### CODE REVIEW

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]
fix tests

**Comment by nrjss67** [comment]
> fix tests

but, there are no errors in pycharm when running the tests 

Снимок экрана 2024-04-15 в 16 21 56

File: app/main.py [hunk 1]
```diff
@@ -1,3 +1,41 @@
...
+                              customer["location"],
+                              customer["money"],
+                              Car(*customer["car"].values()))
+            customers_list.append(person)
+        shops_list = [Shop(*shop.values()) for shop in config_dict["shops"]]
+
+    for customer in customers_list:
+        customer.money_available()
+        home = customer.location
+        min_cost = min(customer.trip_costing(shops_list, fuel_cost),
```
Comment (vasylhnatiuk) [inline app/main.py]: min_cost is not clear name, rename it pls

File: app/main.py [hunk 2]
```diff
@@ -1,3 +1,41 @@
...
+from app.customer import Customer
+from app.car import Car
+from app.shop import Shop
+
+
+def shop_trip() -> None:
+
+    with open("app/config.json", "r") as config:
+        config_dict = json.load(config)
+        fuel_cost = config_dict["FUEL_PRICE"]
```
Comment (vasylhnatiuk) [inline app/main.py]: don't write this and belllow code in `with open("app/config.json", "r") as config:`

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: app/shop.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        self.name = name
+        self.location = location
+        self.products = products
+
+    def print_check(self, name: str, cart: dict) -> None:
+        total_costs = 0
+        test_date = datetime(2021, 1, 4, 12, 33, 41)
+        print(f"Date: {test_date.strftime('%d/%m/%Y %H:%M:%S')}")
+        print(f"Thanks, {name}, for your purchase!")
+        print("You have bought:")
```
Comment (vsmutok) [inline app/shop.py]: combine into one `print()`

File: app/shop.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        for product, amount in cart.items():
+            cost = amount * self.products[product]
+            print(f"{amount} {product}s for {cost} dollars".replace(".0", ""))
+            total_costs += cost
+        print(f"Total cost is {total_costs} dollars")
+        print("See you again!\n")
```
Comment (vsmutok) [inline app/shop.py]: combine into one `print()`

File: app/main.py [hunk 5]
```diff
@@ -1,3 +1,41 @@
...
+        home = customer.location
+        trip_min_cost = min(customer.trip_costing(shops_list, fuel_cost),
+                            key=lambda x: x[2])
+        if trip_min_cost[2] <= customer.money:
+            print(f"{customer.name} rides to {trip_min_cost[1]}\n")
+            customer.location = trip_min_cost[0].location
+            trip_min_cost[0].print_check(customer.name, customer.product_cart)
+            print(f"{customer.name} rides home")
+            print(f"{customer.name} now has "
+                  f"{round(customer.money - trip_min_cost[2], 2)} dollars\n")
```
Comment (vsmutok) [inline app/main.py]: combine into one `print()`

**Review by vsmutok — CHANGES_REQUESTED** [review]

File: app/customer.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        self.car = car
+
+    def money_available(self) -> str:
+        print(f"{self.name} has {self.money} dollars")
+
+    def distance_calculation(self, shop: Shop) -> float:
+        return sqrt((shop.location[0]
+                    - self.location[0])
+                    ** 2 + (shop.location[1]
+                    - self.location[1]) ** 2)
```
Comment (vsmutok) [inline app/customer.py:28]: you can use `math.dist`

**Review by vsmutok — APPROVED** [review]

**Pull request closed without merging**